pkgs/development/tools/click/default.nix

```diff
@@ -0,0 +1,90 @@
+{ lib
+, fetchFromGitLab
+, buildPythonPackage
+, autoconf
+, debian
+, automake
+, perl
+, libtool
+, vala
+, pkg-config
+, libgee
+, json-glib
+, properties-cpp
+, gobject-introspection
+, getopt
+, setuptools
+, pygobject3
+, wrapGAppsHook
+}:
+
+buildPythonPackage {
+  pname = "click";
+  version = "22-02-2023";
+
+  src = fetchFromGitLab {
+    owner = "ubports";
+    repo = "development/core/click";
+    rev = "aaf2735e8e6cbeaf2e429c70136733513a81718a";
+    sha256 = "sha256-pNu995/w3tbz15QQVdVYBnWnAoZmqWj1DN/5PZZ0iZw=";
+  };
+
+
+  configurePhase = ''
+    ./autogen.sh
+    ./configure --prefix=$out \
+        --with-systemdsystemunitdir=/lib/systemd/system \
+        --with-systemduserunitdir=/usr/lib/systemd/user
+  '';
+
+  buildPhase = ''
+    make
+  '';
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r {bin,click_package,lib} $out
+  '';
```

Comment:
This is overriding phases without calling their `pre` and `post` hooks, and calling `make` without checking for & using parallelisation. If you use `format = "other"`, then most of this should already be working without overriding the phases. You should only have to add i.e. `configureFlags`, some packages & build hooks and maybe export some variables.